app/src/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -465,7 +465,12 @@ public void onClick(View v) {
                 
                 //csims@dimagi.com - Jan 24, 2012
                 //Since these are parceled across the content resolver, there's no guarantee of reference equality.
-                //We need to manually check value equality on the type 
+                //We need to manually check value equality on the type
+
+                if(uri == null) { // COMMCARE-163637: might be null if we're using the app in "awesome mode" - in which case we just get out
```

Comment:
I'm not super comfortable with this change. Why does this return null in awesome mode?